site/en/blog/web-custom-formats-for-the-async-clipboard-api/index.md

```diff
@@ -0,0 +1,96 @@
+subtitle: >
+  Web custom formats let websites read and write arbitrary unsanitized payloads using a standardized
+  format applications can opt in to if they wish to support such payloads.
+authors:
+  - thomassteiner
+date: 2022-07-25
+---
+
+Up until now, the [Async Clipboard API](/async-clipboard/) supported a limited set of MIME types to
+be copied to and pasted from the system clipboard. Namely, these types are `text/plain`,
+`text/html`, and `image/png`, which the browser typically will sanitize, for example, to remove
+embedded `script` elements or `javascript:` links from an HTML string, or to prevent PNG
+[decompression bomb](https://en.wikipedia.org/wiki/Zip_bomb) attacks.
+
+In some cases, though, it can be desirable to support unsanitized content on the clipboard, for
+example, when the application the data is thought for deals itself with the sanitization and where
+it's crucial for the copied data to be identical with the pasted data. For such cases, the Async
+Clipboard API now supports web custom formats that let developers write arbitrary data to the
+clipboard.
+
+## Browser support
+
+The Async Clipboard API per se with image support is supported as of Chromium&nbsp;76. Custom
+formats for the Async Clipboard API are supported on desktop and on mobile Chromium as of
+version&nbsp;104.
+
+## Writing web custom formats to the clipboard
+
+Writing web custom formats to the clipboard is almost identical to
+[writing sanitized formats](<https://web.dev/async-clipboard/#write()>), except for the requirement
+to prepend the string `"web "` (including the trailing space) to the blob's MIME type.
+
+```js
+// Fetch remote JPEG and GIF images and obtain their blob representations.
+const jpegBlob = await fetch('image.jpg').then((response) => response.blob());
+const gifBlob = await fetch('image.gif').then((response) => response.blob());
+
+try {
+  // Write the image data to the clipboard, prepending the blobs' actual
+  // types (`"image/jpeg"` and "image/gif") with the string `"web "`, so
+  // it becomes `"web image/jpeg"` and `"web image/gif"` respectively.
+  const clipboardItem = new ClipboardItem({
+    [`web ${jpegBlob.type}`]: jpegBlob,
+    [`web ${gifBlob.type}`]: gifBlob,
+  });
+  await navigator.clipboard.write([clipboardItem]);
+} catch (err) {
+  console.error(err.name, err.message);
+}
+```
+
+## Reading web custom formats from the clipboard
+
+Similar to writing, reading web custom formats from the clipboard is almost identical to
+[reading sanitized formats](<https://web.dev/async-clipboard/#read()>). The only difference is that
+the app now needs to look for clipboard items whose type starts with `"web "`.
+
+```js
+try {
+  // Iterate over all clipboard items.
+  const clipboardItems = await navigator.clipboard.read();
+  for (const clipboardItem of clipboardItems) {
+    for (const type of clipboardItem.types) {
+      // Discard any types that are not web custom formats.
+      if (!type.startsWith('web ')) {
```

Comment:
Does it make sense to highlight lines 68-69?

Reply:
Done.